platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/MemoryIndexDiskKeyValueStore.java

```diff
@@ -140,40 +146,48 @@ public MemoryIndexDiskKeyValueStore(
     }
 
     /**
-     * Merge all files that match the given filter
+     * Compact (merge) all files that match the given filter.
      *
+     * @param clock clock to use to get current time
+     * @param start the moment when compaction is started
      * @param filterForFilesToMerge filter to choose which subset of files to merge
-     * @param minNumberOfFilesToMerge The minimum number of files to consider for a merge
+     * @param minNumberOfFilesToMerge the minimum number of files to consider for a merge
+     * @param reportDurationMetricFunction function to report how long compaction took, in ms
+     * @param reportSavedSpaceMetricFunction function to report how much space was compacted, in Mb
+     * @return the moment when compaction is finished
      * @throws IOException if there was a problem merging
      * @throws InterruptedException if the merge thread was interupted
      */
-    public void merge(
+    public Instant merge(
```

Comment:
I don't think making time measurement part of the API is a good idea.

Reply:
It doesn't have to be. There was a comment in merge() that creating `Instant` instances was expensive, and there was code to save a few redundant objects. I kept that idea. If it isn't really expensive, or if `System.currentTimeMillis()` resolution is enough (time is reduced to millis anyway), I can just call `merge()` and check how long it takes.

Reply:
If you use `Instant.of(seconds, nanos)` creation is very cheap (basically, check/fix range for nanos and assign two attributes).  It's only if you call Instant.now() (which localizes) or any of the other localized calls that it can get more expensive.

Reply:
Removed Clock / Instant args and return value. Now the methods capture merge duration on their own using `System.currentTimeMillis()`